torch/distributed/distributed_c10d.py

```diff
@@ -483,7 +488,10 @@ def init_process_group(backend,
     # barrier at the end to ensure that once we return from this method, all
     # process groups including global variables are updated correctly on all
     # ranks.
-    barrier()
+    if deviceId is not None and backend != Backend.NCCL:
+        # Ignore deviceId for backend other than NCCL
+        deviceId = None
```

Comment:
Nit: If we do go with this approach, would be nice to have a warning here that the parameter is not being used and the user does not have to specify it.

Reply:
> Is there any backward compatibility issues that we should be aware of?

The device id will be optional. So barrier() will continue to support default rank to device mapping.